### PR TITLE
Feat/tour templates

### LINKED ISSUE #966: [BUG] Tours can't read from data lists
**Describe the bug**
Tours can't read from data lists: when using a data list item in a tour, it displays as **undefined**. 

**App version**
v0.11.1 (and before)

**Debug sheets**
template: [debug_tour_template ](https://docs.google.com/spreadsheets/d/1CQzsurOr4egC706cx_Fvkk5Zfvl-vRmd-Pu40mYM1qA/edit#gid=1916689711)
tour: [debug_tour](https://docs.google.com/spreadsheets/d/1CQzsurOr4egC706cx_Fvkk5Zfvl-vRmd-Pu40mYM1qA/edit#gid=95491335)

**To reproduce**
Launch debug_tour_template (note that the data item prints correctly here). Click the button to start the debug_tour. The second step in the tour shows the data item not working. 

**Screenshots**
image
image

### LINKED ISSUE #967: Tour improvements
**Details**

- [ ] #969
- [ ]  Tour sheets are currently written in HTML (they are in fact the only sheet that use HTML). We should probably update the authoring syntax to be in line with the rest of the app spreadsheets.

**Screenshots and videos**
image

### PR DESCRIPTION
**Pull request by chrismclarke**
PR Checklist

- [x] - Latest `master` branch merged
- [x] - PR title descriptive (can be used in release notes)
## Breaking Changes
- Added new 3rd party dependencies, so require `yarn install`

## Description
One of the limitations with the tour system is a lack of support for the rest of template syntax, such as use of `@global`, `@data` etc. One potential workaround discussed was to allow tours to directly render a template within their content, to allow the existing templating system to handle logic.

This PR adds that support, with the inclusion of a `message_template` column that will be used to render a template in a tour step. Note this is still somewhat limited as individual templates are required for each tour step (instead of closer integration within a single template, such as `begin_tour_step` and `end_tour_step` components), but the solution is likely a precursor for future changes (until we're ready to replace out intro.js with our own system).

This is because the way the `intro.js` system handles rendering content is done in a way that would bypass what angular recognises as rendered, and naturally would only support html content that the web browser natively understands (e.g. `<div>` `<span>` etc.). In order to enable intro.js to render templates an additional set of tooling called [Angular Elements](https://angular.io/guide/elements) is required which provides the definitions required so that the webpage can natively render specifically the `template-container` component, making the required link with angular.

Main changes:

- Add support for angular custom elements
- Create and register custom element for template containers
- Create template [example_tour_templates](https://docs.google.com/spreadsheets/d/1CQzsurOr4egC706cx_Fvkk5Zfvl-vRmd-Pu40mYM1qA/edit#gid=746362818) to demo functionality

Note - Type definition updates required have already been merged in #1007 so that updated examples can be synced without breaking build.

## Git Issues

Closes #967 
Closes #966

## Screenshots/Videos

_example_tour_templates_
image

_example_tour_template_1_
image

Note - to access the list of tours for testing you will likely need to manually go to the `/tour` page, e.g. http://localhost:4200/tour

https://user-images.githubusercontent.com/10515065/132409843-2453d939-69cb-4901-8867-415f0c7710b9.mp4

### CODE REVIEW

File: src/app/shared/services/tour/tour.service.ts [hunk 1]
```diff
@@ -45,20 +50,27 @@ export class TourService {
...
+          let stepContent = this.hackReplaceGlobalInRowMessage(translatedRow.message_text);
+          // if using a template for the step content load using the custom template-container web-component
+          if (translatedRow.message_template) {
+            stepContent = `
+              <web-template-container templatename=${translatedRow.message_template}></web-template-container>
+            `;
+          }
           return {
-            intro: this.hackReplaceGlobalInRowMessage(translatedRow.message_text),
-            title: this.hackReplaceGlobalInRowMessage(translatedRow.title),
```
Comment (volloholic) [inline src/app/shared/services/tour/tour.service.ts:50]: I didn't realise these were already in.

**Review by volloholic — APPROVED** [review]
The code looks fine.
@EsmeeIDEMS Could you do a functional test review and then merge.

File: src/app/shared/services/tour/tour.service.ts (hunk 1 above)
Comment (chrismclarke) [reply src/app/shared/services/tour/tour.service.ts:50]: Yes, although it is only globals supported. Which is why we still could have added a short term solution to just manually add methods `data` and some others (and still could do), but hopefully being able to embed templates directly is also a good option

**Review by EsmeeIDEMS — APPROVED** [review]
Functional test passed

**Pull request merged**